openlcs/libs/corgi.py

```diff
@@ -379,8 +380,20 @@ def get_container_source_components(self, component, subscribed_purls=None,
         if name.endswith("-source"):
             logger.debug("This is a source container build")
             sources = component.get("sources")
+            # There are cases when a "-source" components in corgi does not
+            # have a corresponding binary build. See also OLCS-459
             if not sources:
-                return None
+                # Remove trailing "-sources" from name
+                binary_build_name = re.sub(r"-source$", "", name)
+                nevra = component.get("nevra")
+                # Get nevra for corresponding binary build
+                binary_nevra = re.sub(
+                    f"({binary_build_name})-source", r"\1", nevra)
```

Comment:
This could be wrong. In brew the `-source` build may have trailing incremental number in its nvr.

Reply:
Latest code get rid of the error prone guessing. Also added test case to cover the situation.